src/braket/experimental/autoqasm/api.py

```diff
@@ -60,15 +60,15 @@ def main(*args, num_qubits: Optional[int] = None) -> Callable[[Any], aq_program.
         #    @aq.main
         #    def my_func(...):
         # Equivalently, `main(my_func, num_qubits=None)`
-        return _function_wrapper(args[0], user_config=user_config)
+        return _function_wrapper(args[0], _convert_main, wrapper_args={"user_config": user_config})
     else:
         # In this case, num_qubits was supplied, and we don't know `f` yet.
         # Matches the following syntax:
         #    @aq.main(num_qubits=x)
         #    def my_func(...):
         # Equivalently: `main(num_qubits=x)(my_func)`
         def _function_with_params(f: Callable) -> Callable[[Any], aq_program.Program]:
-            return _function_wrapper(f, user_config=user_config)
+            return _function_wrapper(f, _convert_main, wrapper_args={"user_config": user_config})
 
         return _function_with_params
```

Comment:
Not directly related to your changes, but I realized when doing the `gate()` decorator that the implementation of `main()` could be simplified a little bit:

```
def main(*args, num_qubits: Optional[int] = None) -> Callable[[Any], aq_program.Program]:
   """docstring here"""
    if not args:
        # Only keyword args were passed to the decorator, so we don't know `f` yet.
        # Matches the following syntax:
        #    @aq.main(num_qubits=x)
        #    def my_func(...):
        # Equivalently: `aq.main(num_qubits=x)(my_func)`
        def _function_with_params(*args) -> Callable[[Any], aq_program.Program]:
            return main(*args, num_qubits=num_qubits)
        return _function_with_params

    user_config = aq_program.UserConfig(num_qubits=num_qubits)
    return _function_wrapper(args[0], _convert_main, wrapper_args={"user_config": user_config})
```

Reply:
In fact, I wonder if it would be possible to move that `if not args:` block into `_function_wrapper` as well? This would also make it so that, for example, `@aq.subroutine` and `@aq.subroutine()` always behave equivalently, for all of our decorators - which could potentially be a paper cut. Then `main()` could end up being just two lines:
```
    user_config = aq_program.UserConfig(num_qubits=num_qubits)
    return _function_wrapper(args[0], _convert_main, wrapper_args={"user_config": user_config})
```

Reply:
I couldn't get it to work inside _function_wrapper, but I could try again as a follow up

Reply:
For now I deleted the comments and removed the "else"